### PR TITLE
feat: add cloud agent

### PR DESCRIPTION
**Pull request by povilasv**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: cmd/api-server/main.go [hunk 1]
```diff
@@ -255,6 +256,26 @@ func main() {
 		scheduler,
 	)
 
+	tkcEnabled := os.Getenv("TESTKUBE_CLOUD_ENABLED")
```
Comment (exu) [inline cmd/api-server/main.go:259]: Will it be enough to just check if there is an API key?

**Review by exu — APPROVED** [review]
Small comment

File: pkg/agent/agent.go [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	handler fasthttp.RequestHandler
+	logger  *zap.SugaredLogger
+	apiKey  string
+}
+
+func NewAgent(logger *zap.SugaredLogger, handler fasthttp.RequestHandler, server, apiKey string, isInsecure bool) (*Agent, error) {
+
+	var security grpc.DialOption
+	security = grpc.WithTransportCredentials(credentials.NewTLS(nil))
+	if isInsecure {
```
Comment (vsukhin) [inline pkg/agent/agent.go:30]: I would use if else, if we defined security var

File: pkg/agent/agent.go [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+		return nil, err
+	}
+
+	client := cloud.NewTestKubeCloudAPIClient(conn)
+	return &Agent{conn: conn, client: client, handler: handler, logger: logger, apiKey: apiKey}, nil
+}
+
+func (ag *Agent) Run(ctx context.Context) error {
+	var opts []grpc.CallOption
+	md := metadata.Pairs("api-key", ag.apiKey)
```
Comment (vsukhin) [inline pkg/agent/agent.go:44]: might be define a const for api-key

File: pkg/agent/agent.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			ag.logger.Errorf("agent stream recv: %w", err)
+			return err
+		}
+
+		switch {
+		case cmd.Url == "healthcheck":
```
Comment (vsukhin) [inline pkg/agent/agent.go:67]: should we check ctx.Done()?

Comment (vsukhin) [inline pkg/agent/agent.go:67]: should it be const?

File: pkg/agent/agent.go [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+					ag.logger.Errorf("stream send: %w", err)
+				}
+				return err
+			}
+			r.SetURI(uri)
+
+			req.Init(r, nil, nil)
+			ag.handler(req)
+
+			fasthttp.ReleaseRequest(r)
```
Comment (vsukhin) [inline pkg/agent/agent.go:102]: no errors in all statements above?

File: pkg/agent/agent.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+			headers := make(map[string]*cloud.HeaderValue)
+			req.Response.Header.VisitAll(func(key, value []byte) {
+				_, ok := headers[string(key)]
+				if !ok {
+					headers[string(key)] = &cloud.HeaderValue{Header: []string{string(value)}}
+					return
```
Comment (vsukhin) [inline pkg/agent/agent.go:109]: what do we return there?

**Review by vsukhin — APPROVED** [review]
looks good, some minor notes

**Pull request merged**